src/Plutip/Server.purs

```diff
@@ -216,8 +230,49 @@ type CtlServerConfig =
   , networkId :: NetworkInfo
   }
 
-runCtlServer :: CtlServerConfig -> Aff Unit
-runCtlServer = undefined
+startCtlServer :: PlutipConfig -> ClusterStartupParameters -> Aff ChildProcess
+startCtlServer cfg params = do
+  networkInfo <- liftEffect $ getNetworkInfoFromNodeConfig ""
+  let
+    ctlServerArgs =
+      [ "--port"
+      , UInt.toString cfg.ctlServerConfig.port
+      , "--node-socket"
+      , params.nodeSocketPath
+      ] <> getNetworkInfoArgs networkInfo
+  liftEffect $ spawn "ctl-server" ctlServerArgs defaultSpawnOptions
+  where
+  getNetworkInfoArgs :: NetworkInfo -> Array String
+  getNetworkInfoArgs Mainnet = []
+  getNetworkInfoArgs (Testnet networkId) =
+    [ "--network-id", Int.toStringAs Int.decimal networkId ]
+
+stopCtlServer :: ChildProcess -> Aff Unit
+stopCtlServer = liftEffect <<< kill SIGINT
+
+-- eihter' :: (a -> c) -> Either a b -> (b -> c) -> c
+
+getNetworkInfoFromNodeConfig :: FilePath -> Effect NetworkInfo
+getNetworkInfoFromNodeConfig nodeConfigPath = do
+  nodeConfigJson <- either (\e -> throw $ show e) pure =<<
+    parseJsonStringToAeson <$> readTextFile Encoding.UTF8 nodeConfigPath
+  let
+    byronGenesisFile = getField
+      (unsafePartial $ fromJust $ toObject nodeConfigJson)
```

Comment:
We should not use `unsafePartial`. Please use `fromMaybe (throw "some error message")` - all possible failure points must be explicitly handled. I think you can also parse everything is one go in monadic context. Also please use `.:` (a synonym for `getField`)

Reply:
removed